CHANGELOG.md

```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+- New BOS v2 option `session_limit_required`
+  - If set, new sessions must have a limit specified
+  - When creating a BOS session, specifying a limit value of `*` has the effect of not limiting
```

Comment:
Yeah -- I actually kind of considered that to be an advantage, because it made it even harder to do by accident. The only time you'd need to be specifying it is when you've got the require limit option enabled -- otherwise there'd be no point in specifying it. I guess one downside is that if they specify * without quotes, it won't result in a blank value, but will instead be the files in their current directory. But unless there is just a single file, that would result in a syntax error with the CLI. If it was a single file in the directory, then its name would be interpreted as the limit. Most likely the file name would not match an xname or an HSM group or role, so most likely the result would be a session that ends up acting on 0 nodes. Which isn't the worst thing.

The other advantage is that no code changes were required to support '*', because that is already a valid limit. It wouldn't require a big code change to support all, but it would be slightly more invasive (more so if we allow `all` to also appear in a comma-separated limit list).

I don't feel strongly about it, so if you do, I'm okay making that change.

Reply:
I think you'd have bad luck to have file names happening to match xnames. It's not out of the realm of possibility, but you'd need both those files to exist and to forget to quote the asterisks. It would suck though to think you'd rebooted the entire system only to find out later that nope you tried to reboot node 'random_file.txt'. And you wouldn't figure that out until you checked on your session which should have ended prematurely with '0 nodes to operate on'. If it was immediate feedback saying, "No, bad user," I'd be more happy with it.

I'm not happy to hear that 'all' is more invasive. For an option I didn't care that much about supporting in the first place, I don't know that it's worth jeopardizing/destabilizing the code more just to use 'all' instead of '*'.

Reply:
Well, if you looked at the response from BOS when you'd created the session, you may noticed that the limit was set to `random_file.txt` instead of `*`.

And in my experience, the admins almost immediately start watching the status of their BOS sessions after they create them. So I don't think it would take very long for them to notice something was up.

That said, the `all` option isn't SUPER invasive. It just requires more code changes, since `*` is already a valid limit, even before my PR. So I didn't have to add any code to handle it. Whereas I would have to add code to interpret `all`. But it would really just amount to having BOS treat `all` the same way that it currently treats `*`. But I figured that since this option is disabled by default and customers who enable it are presumably ones who usually intend to specify limits for their sessions, it seemed a worthwhile tradeoff to use `*`, despite the requirement of quoting it on the command line.